spec/text/generator_spec.rb

```diff
@@ -1,146 +1,151 @@
 require 'rails_helper'
 
 RSpec.describe Text::Generator do
-  let(:report)         { create(:report) }
-  let(:intention)      { :real }
+  let!(:sensorstory_channel)  { create(:channel, name: "sensorstory", report: report) }
+  let(:report)                { create(:report) }
+  let(:intention)             { :real }
   subject { described_class.new(report: report, opts: {intention: intention}) }
 
-  describe '#choose_heading' do
-    subject { super().choose_heading }
-    let(:report) { create(:report, text_components: text_components) }
+  context 'for text components with the same report and the same channel' do
```

Comment:
We are testing this situation and only this situation. We cannot say what happens for any other combination. If we really want to stick to static data in the database, let's make `Channel` a static read-only model and generate the two entries with a migration. I dislike seed data in migrations, but actually that wouldn't be seed data, because we expect it in our tests as well.